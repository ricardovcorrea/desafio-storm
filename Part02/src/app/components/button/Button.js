import React from "react";
import "./Button.scss";

import PropTypes from "prop-types";

import Icon from "@material-ui/core/Icon";
import Button from "@material-ui/core/Button";

const CustomButton = props => {
  const iconClass = props.text ? "iconWithText" : "";

  return (
    <Button
      variant="contained"
      className={"button"}
      style={{ backgroundColor: props.bgColor ? props.bgColor : "#fff" }}
      onClick={props.onClick}
    >
      {props.icon && (
        <Icon
          className={iconClass}
          style={{ color: props.color ? props.color : "" }}
        >
          {props.icon}
        </Icon>
      )}
      <span style={{ color: props.color ? props.color : "" }}>
        {props.text || ""}
      </span>
    </Button>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  text: PropTypes.string,
  bgColor: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func
};

export default CustomButton;
