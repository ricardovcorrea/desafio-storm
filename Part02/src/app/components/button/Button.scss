.button {
    margin: 5px !important;
    height: 50px;
}
.iconWithText {
    margin-right: 5px;
}