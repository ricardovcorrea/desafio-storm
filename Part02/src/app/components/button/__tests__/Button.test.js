import React from 'react';
import { render } from '@testing-library/react';
import Button from '../Button';

test('renders button', () => {
  const container = render(<Button icon="settings" text="teste"/>)
  expect(container.container).toMatchSnapshot()
});
