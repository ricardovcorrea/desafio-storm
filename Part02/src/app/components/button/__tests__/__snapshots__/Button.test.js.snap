// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained button"
    style="background-color: rgb(255, 255, 255);"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root iconWithText"
      >
        settings
      </span>
      <span>
        teste
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
